multiversxetl/indexer.py

```diff
@@ -0,0 +1,67 @@
+from typing import Any, Dict, Iterable
+
+import elasticsearch.helpers
+from elasticsearch import Elasticsearch
+
+SCROLL_CONSISTENCY_TIME = "10m"
+SCAN_BATCH_SIZE = 9999
+
+
+class Indexer:
+    def __init__(self, url: str):
+        self.elastic_search_client = Elasticsearch(url)
+
+    def count_records_with_interval(self, index_name: str) -> int:
+        return self.elastic_search_client.count(index=index_name)["count"]
+
+    def count_records_without_interval(self, index_name: str) -> int:
+        return self.elastic_search_client.count(index=index_name)["count"]
+
+    def get_records_with_interval(self, index_name: str, start_timestamp: int, end_timestamp: int) -> Iterable[Dict[str, Any]]:
+        query = {
+            "query": {
+                "range": {
+                    "timestamp": {
+                        "gte": start_timestamp,
+                        "lte": end_timestamp,
+                    },
+                }
+            }
+        }
+
+        records = elasticsearch.helpers.scan(
+            client=self.elastic_search_client,
+            index=index_name,
+            query=query,
+            scroll=SCROLL_CONSISTENCY_TIME,
+            raise_on_error=True,
+            preserve_order=False,
+            size=SCAN_BATCH_SIZE,
+            request_timeout=None,
+            scroll_kwargs=None,
+            clear_scroll=True
+        )
+
+        return records
+
+    def get_records_without_interval(self, index_name: str) -> Iterable[Dict[str, Any]]:
+        query: Any = {
+            "query": {
+                "match_all": {},
+            }
+        }
+
+        records = elasticsearch.helpers.scan(
```

Comment:
instead of this, for RAM usage optimizations, please refactor so you save each bulk at a time instead of gathering all the bulks in a single variable. 

example: https://github.com/multiversx/mx-chain-tools-go/blob/c8b2819dea9a035d81fe4084b0f76bcea126c845/elasticreindexer/elastic/elasticClient.go#L226

Reply:
also, regarding scroll duration, make sure you change it for each request because the response can be cached if the request is the same. for example, in the aforementioned code, the duration is different each time

Reply:
Very good point!

Added this issue: https://github.com/multiversx/multiversx-etl/issues/9